tests/test_models.py

```diff
@@ -253,6 +253,41 @@ def test_user_cpu_usage(db, session, new_user, run_workflow):
     assert new_user.get_quota_usage()["disk"]["usage"]["raw"] == 128
 
 
+@mock.patch(
+    "reana_db.utils.PERIODIC_RESOURCE_QUOTA_UPDATE_POLICY", True,
+)
+@pytest.mark.parametrize(
+    "WORKFLOW_TERMINATION_QUOTA_UPDATE_POLICY",
+    [([QuotaResourceType.cpu.value]), ([]),],
+)
+def test_all_users_cpu_quota_usage_update(
+    db, session, new_user, run_workflow, WORKFLOW_TERMINATION_QUOTA_UPDATE_POLICY
+):
+    """Test CPU periodic update cronjob functionality.
+
+    Multiple scenarios are tested:
+    - CPU periodic update enabled + CPU workflow termination update policy enabled
+    - CPU periodic update enabled + CPU workflow termination update policy disabled
+    """
+    with mock.patch(
+        "reana_db.utils.WORKFLOW_TERMINATION_QUOTA_UPDATE_POLICY",
+        WORKFLOW_TERMINATION_QUOTA_UPDATE_POLICY,
+    ):
+        time_elapsed_seconds = 0.5
+        num_workflows = 2
+        for n in range(num_workflows):
+            run_workflow(time_elapsed_seconds=time_elapsed_seconds)
+
+        if not WORKFLOW_TERMINATION_QUOTA_UPDATE_POLICY:
+            # verify that workflow termination policy logic is not executed
+            assert new_user.get_quota_usage()["cpu"]["usage"]["raw"] == 0
+        update_users_cpu_quota()
+        assert (
+            new_user.get_quota_usage()["cpu"]["usage"]["raw"]
+            >= num_workflows * time_elapsed_seconds * 1000
+        )
```

Comment:
What do you think?

Reply:
*thought:* in general, the quota code is hard to understand. `update_users_cpu_quota` and other quota functions that rely on checking of env variables inside, have non-obvious behavior. For example, `update_users_cpu_quota` has a verb `update_...` which implies it will update CPU quotas but sometimes it will not if combination of env variables is not right.

This is not a good long-term decision to put ENV checks inside (hello to me). 

*thought 2:* `WORKFLOW_TERMINATION_QUOTA_UPDATE_POLICY` is used not only to configure what quotas are calculated when the workflow is finished but also affects whatever quotas are calculated in the REST API endpoint in `r-workflow-controller` during file upload or deletion ([line](https://github.com/reanahub/reana-workflow-controller/blob/a3e2af49a4c8153a46cee939ff74cfb81ae2c356/reana_workflow_controller/rest/workflows_workspace.py#L157)). And this is not obvious from the name, `...TERMINATION...` don't imply it will stop calculating quotas during upload/deletion. From the user/admin perspective, maybe, not much to worry about but as a developer, it gives me a headache.

*suggestion:* I can create an issue concerning the above problems. Those inconsistencies were introduced because we wanted to quickly disable disk quotas to benchmark cluster but I am not sure it is a long-term solution. What do you think? @tiborsimko

Reply:
Concerning `test_user_cpu_usage`, I think `periodic_update` is not relevant, as this test function is only checking the termination policy logic, i.e. the workflow status event listener. By default `PERIODIC_RESOURCE_QUOTA_UPDATE_POLICY` is false so it's irrelevant. Actually, I think it's better not to include `PERIODIC_RESOURCE_QUOTA_UPDATE_POLICY` to verify that our default values are right. Said that, I also believe the test name can be better, something like `test_workflow_termination_user_quota_usage` might be better.

Concerning `test_all_users_cpu_quota_usage_update`, I think it's also useful to run some asserts before the `update_users_cpu_quota()` call to verify that that quota has/has not been calculated, otherwise, the CPU updates might have been executed before and the asserts _a posteriori_ are not as reliable.